testutil/fakegraphsync.go

```diff
@@ -422,6 +422,11 @@ func (fgs *FakeGraphSync) Stats() graphsync.Stats {
 	return graphsync.Stats{}
 }
 
+func (fgs *FakeGraphSync) RegisterOutgoingRequestProcessingListener(graphsync.OnOutgoingRequestProcessingListener) graphsync.UnregisterHookFunc {
```

Comment:
where this came from why this exist

Reply:
yea that's a mock GraphExchange interface that needs a new method cause Graphsync updated and added it.. I'm moving away from putting methods on GraphExchange and towards only implementing them on the main implementation to simplify this. But later.